src/core/batchlan2.cpp

```diff
@@ -2167,21 +2167,14 @@ bool    _ElementaryCommand::ConstructBGM (_String&source, _ExecutionList&target)
 
 //____________________________________________________________________________________
 
-void    RetrieveModelComponents (long mid, _Matrix*& mm, _Matrix*& fv, bool & mbf)
-{
-    if (mid >=0 && mid < modelTypeList.lLength) {
-        if (modelTypeList.lData[mid] == 0) {
-            mm = (_Matrix*)FetchObjectFromVariableByTypeIndex(modelMatrixIndices.lData[mid],MATRIX);
-        } else {
-            mm = nil;
-        }
-
-        long fvi = modelFrequenciesIndices.lData[mid];
-        fv = (_Matrix*)FetchObjectFromVariableByTypeIndex(fvi>=0?fvi:(-fvi-1),MATRIX);
-        mbf = (fvi>=0);
-    } else {
-        mm = fv = nil;
-        mbf = false;
+void    RetrieveModelComponents (long mid, _Matrix*& mm, _Matrix*& fv, bool & mbf) {
```

Comment:
There is an overloaded `RetrieveModelComponents` function that was not updated. Will need to review whether that function was affected as well.

Reply:
Let me check to see if this is correct

Reply:
The `RetrieveModelComponents (long mid, _Variable*& mm, _Variable*& fv, bool & mbf)` function is correct, because it uses `LocateVar` which does not evaluate matrix valued expressions (which was an issue with the changed function)

Reply:
I see it now. Thank you for the additional information.